src/app/dashboard/student-preview-card/student-preview-card.component.ts

```diff
@@ -0,0 +1,64 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { Student, Device } from 'src/app/api/models';
+import { ProjectsService } from 'src/app/shared/data/projects.service';
+import { GenderService } from 'src/app/shared/helpers/gender.service';
+import { GravatarService } from 'src/app/shared/helpers/gravatar.service';
+import { NationalityService } from 'src/app/shared/helpers/nationality.service';
+
+@Component({
+  selector: 'app-student-preview-card',
+  templateUrl: './student-preview-card.component.html',
+  styleUrl: './student-preview-card.component.scss',
+})
+export class StudentPreviewCardComponent implements OnInit {
+  @Input({ required: true }) student: Student;
+  @Input() projectId: string;
+  Device = Device;
+  private readonly projectPreferenceLimit = 4;
+
+  constructor(
+    private nationalityService: NationalityService,
+    private projectsService: ProjectsService,
+    private genderService: GenderService,
+    private gravatarService: GravatarService
+  ) {}
+
+  ngOnInit() {
+    this.projectPreferences = this.getProjectPreferences().slice(0, this.projectPreferenceLimit);
+    this.projectPreferenceScore = this.getProjectPreferenceScore();
+    this.germanProficiency = this.student.languages.find(l => l.language === 'de').proficiency;
```

Comment:
I would also put this in a separate function. 
 ` this.germanProficiency = this.findGermanProficiency() `
 
```
 findGermanProficiency() {
  return  this.student.languages.find(l => l.language === 'de').proficiency;
  }
```